### ISSUE TITLE
Vertical effect in spherical camera image

### ISSUE THREAD

**Issue by stefaniapedrazzi** [cyberbotics/webots#2295]
A user asked for vertical effect in spherical image because currently only horizontal effect is applied:

![horizontal_effect](https://user-images.githubusercontent.com/5910449/95092488-fe824000-0727-11eb-8a9f-7a3322a48218.png)
![vertical_effect](https://user-images.githubusercontent.com/5910449/95092493-ffb36d00-0727-11eb-8d96-fd478ded251d.png)

It seems correct to also expect a vertical spherical effect.
So we should maybe evaluate:
* if it is interesting to add a vertical effect
* if `spherical` is the best description for the image effect or if we could choose a better name (for example `equirectangular` or `cylindric` projection instead of `spherical` projection)

**Comment by omichel** [comment]
Fix this problem would help also Robocup competitors: see https://hl.forum.robocup.org/t/modeling-wideangle-lenses-in-webots/495

**Comment by omichel** [comment]
It seems this issue is not resolved, see https://hl.forum.robocup.org/t/modeling-wideangle-lenses-in-webots/495/7

**Comment by stefaniapedrazzi** [comment]
Related to #3068

**Comment by stefaniapedrazzi** [comment]
To get both vertical and horizontal effect we should use the stereographic projection instead of the equirectangular projection (currently implemented).

**Comment by TheMangalex** [comment]
I'm not sure if this issue is still being worked on, or if there is some finished implementation on the develop already, but in case not, I made my own version of the shader which makes a distortion in both directions, but I'm currently not sure, if it is correctly scaled. (I used a reference implementation as basis and have to recheck the math).
![image](https://user-images.githubusercontent.com/26520790/190665308-bacb5c84-4c59-4e9a-81dc-6342e3099319.png)

Also it seems that replacing the merge_spherical shader is not trivial, as it is also used for the LIDAR projections. Therefore different handling of equirectangular and fisheye projections would be needed.

If you are interested in this fisheye shader, I can share it with you.
